lib/Notification/Notifier.php

```diff
@@ -287,6 +293,62 @@ protected function shortenJsonEncodedMultibyteSave(string $subject, int $dataLen
 		return $temp;
 	}
 
+	private function parseStoredRecording(
+		INotification $notification,
+		Room $room,
+		Participant $participant,
+		IL10N $l
+	): INotification {
+		$parameters = $notification->getSubjectParameters();
+		$shareAction = $notification->createAction()
+			->setParsedLabel($l->t('Share to chat'))
+			->setPrimary(true)
+			->setLink(
+				$this->urlGenerator->linkToRouteAbsolute(
+					'ocs.spreed.Chat.shareObjectToChat',
+					[
+						'apiVersion' => 'v1',
+						'objectType' => 'file',
+						'objectId' => $notification->getObjectId(),
+						'metaData' => json_encode([
+							'name' => $parameters['name'],
+							'path' => $parameters['name'],
+						]),
+						'token' => $room->getToken()
+					]
+				),
+				IAction::TYPE_POST
+			);
+		$dismissAction = $notification->createAction()
+			->setParsedLabel($l->t('Dismiss notification'))
+			->setLink(
+				$this->urlGenerator->linkToRouteAbsolute(
+					'ocs.spreed.Recording.notificationDismiss',
+					[
+						'apiVersion' => 'v1',
+						'token' => $room->getToken(),
+						'dateTime' => $notification->getDateTime()->format('U'),
```

Comment:
Time sounds a bit error prone to me. Why not the file id?

Reply:
The fileId is stored inside a json.
We can remove the dateTime but, is a more specific attribute to prevent dismiss a wrong notification.